### PR TITLE
Gather all Events in RSSAgent before sorting

### LINKED ISSUE #1077: events_order of RSSAgent when defining an array of urls
Hi,
it is possible to define multiple urls for the RSSAgent. In this case the RSS urls are parsed one after one. 

However I want to get a misced output for further processing sorted by publication date. Ok, so I've set the events_order key, but it only effects events created by the SAME url. 

Do you have any solution? What am I doing wrong? 
Thanks,
hunhejj

### PR DESCRIPTION
**Pull request by cantino**
May fix #1077.

### CODE REVIEW

**Comment by cantino** [comment]
Quick review @knu or @dsander?

File: app/models/agents/rss_agent.rb [hunk 1]
```diff
@@ -87,34 +87,41 @@ def events_order
...
+      new_events = []
+      max_events = (interpolated['max_events_per_run'].presence || 0).to_i
+
+      urls.each do |url|
+        begin
+          response = faraday.get(url)
+          if response.success?
+            feed = FeedNormalizer::FeedNormalizer.parse(response.body, loose: true)
+            feed.clean! if boolify(interpolated['clean'])
+            new_events += feed_to_events(feed)
```
Comment (knu) [inline app/models/agents/rss_agent.rb]: `new_events.concat feed_to_events(feed)` could be better in memory usage.

File: app/models/agents/rss_agent.rb [hunk 2]
```diff
@@ -87,34 +87,41 @@ def events_order
...
-          if check_and_track(entry_id)
+    def check_urls(urls)
+      new_events = []
+      max_events = (interpolated['max_events_per_run'].presence || 0).to_i
+
+      urls.each do |url|
+        begin
+          response = faraday.get(url)
+          if response.success?
+            feed = FeedNormalizer::FeedNormalizer.parse(response.body, loose: true)
```
Comment (knu) [inline app/models/agents/rss_agent.rb]: I think parser error should be rescued separately from fetch error, but that's out of this PR's scope so I'll tackle that later.

**Comment by knu** [comment]
Looks good!

File: app/models/agents/rss_agent.rb (hunk 1 above)
Comment (cantino) [inline app/models/agents/rss_agent.rb]: Good call.

File: app/models/agents/rss_agent.rb (hunk 2 above)
Comment (cantino) [inline app/models/agents/rss_agent.rb]: Okay.

**Comment by cantino** [comment]
Thanks!

**Pull request merged**